### PR TITLE
Tests for union types

### PR DESCRIPTION
**Pull request by abhinav**
@prashantv @kriskowal

### CODE REVIEW

File: gen/struct_test.go [hunk 1]
```diff
@@ -318,3 +318,133 @@ func TestNestedStructsOptional(t *testing.T) {
...
+			wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{ID: 1, Value: wire.NewValueBinary([]byte{1, 2, 3})},
+			}}),
+		},
+		{
+			testdata.Document{PlainText: stringp("hello")},
+			wire.NewValueStruct(wire.Struct{Fields: []wire.Field{
+				{ID: 2, Value: wire.NewValueString("hello")},
+			}}),
+		},
```
Comment (prashantv) [inline gen/struct_test.go]: what happens when you get an `ID` that's not recognized?

**Comment by prashantv** [comment]
Let's add tests for:
 - no IDs received
 - single unrecognized ID
 - multiple IDs (either both recognized, or only one recognized, or both recognized)

These should probably cause errors?

**Comment by abhinav** [comment]
:+1:

**Comment by abhinav** [comment]
Updated

**Comment by prashantv** [comment]
lgtm

**Pull request merged**